dotnet-authserver/src/TeacherIdentity.AuthServer/Infrastructure/ModelBinding/WebHookMessageTypesModelBinder.cs

```diff
@@ -0,0 +1,32 @@
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using TeacherIdentity.AuthServer.Models;
+
+namespace TeacherIdentity.AuthServer.Infrastructure.ModelBinding;
+
+public class WebHookMessageTypesModelBinder : IModelBinder
+{
+    public Task BindModelAsync(ModelBindingContext bindingContext)
+    {
+        var modelType = bindingContext.ModelType;
+        if (!typeof(WebHookMessageTypes).IsAssignableTo(modelType))
+        {
+            throw new InvalidOperationException($"Cannot bind to {modelType}.");
+        }
+
+        var valueProviderResult = bindingContext.ValueProvider.GetValue(bindingContext.ModelName);
+        var rawValues = valueProviderResult.Values.ToArray();
+        if (rawValues != null)
```

Comment:
Will this ever be `null`? Would a check on `Length != 0` be better?

Reply:
Amended